src/mame/misc/micom_mahjong.cpp

```diff
@@ -0,0 +1,243 @@
+// license:BSD-3-Clause
+// copyright-holders:hap
+/******************************************************************************
+
+マイコン麻雀 - Micom Mahjong, mail-order Mahjong console.
+
+Hardware notes:
+- Z80, 11.0592MHz XTAL
+- 16KB ROM (4*2732), 1KB RAM
+- 2KB ROM (2716) for tiles, 1KB VRAM, 1bpp video
+- 1-bit sound
+
+TODO:
+- map the keypad
+
+******************************************************************************/
+
+#include "emu.h"
+#include "cpu/z80/z80.h"
+#include "sound/dac.h"
+
+#include "emupal.h"
+#include "screen.h"
+#include "speaker.h"
+#include "tilemap.h"
+
+
+namespace {
+
+class mmahjong_state : public driver_device
+{
+public:
+	mmahjong_state(const machine_config &mconfig, device_type type, const char *tag) :
+		driver_device(mconfig, type, tag),
+		m_maincpu(*this, "maincpu"),
+		m_vram(*this, "vram"),
+		m_tilemap(*this, "tilemap"),
+		m_screen(*this, "screen"),
+		m_dac(*this, "dac"),
+		m_inputs(*this, "IN.%u", 0)
+	{ }
+
+	void mmahjong(machine_config &config);
+
+protected:
+	virtual void machine_start() override;
+
+private:
+	required_device<cpu_device> m_maincpu;
+	required_shared_ptr<u8> m_vram;
+	required_device<tilemap_device> m_tilemap;
+	required_device<screen_device> m_screen;
+	optional_device<dac_bit_interface> m_dac;
```

Comment:
Should this be required? The only machine config has it

Reply:
Yeah, copypaste mistake